editor/groups_editor.cpp

```diff
@@ -32,714 +32,681 @@
-	add_group_button->set_disabled(p_new_text.strip_edges().is_empty());
-}
+	tree->clear();
 
-void GroupDialog::_group_renamed() {
-	TreeItem *renamed_group = groups->get_edited();
-	if (!renamed_group) {
-		return;
+	List<Node::GroupInfo> groups;
+	node->get_groups(&groups);
+	groups.sort_custom<_GroupInfoComparator>();
+
+	List<StringName> current_groups;
+	for (const Node::GroupInfo &gi : groups) {
+		current_groups.push_back(gi.name);
 	}
 
-	const String name = renamed_group->get_text(0).strip_edges();
-	for (TreeItem *E = groups_root->get_first_child(); E; E = E->get_next()) {
-		if (E != renamed_group && E->get_text(0) == name) {
-			renamed_group->set_text(0, selected_group);
-			error->set_text(TTR("Group name already exists."));
-			error->popup_centered();
-			return;
+	TreeItem *root = tree->create_item();
+
+	TreeItem *local_root = tree->create_item(root);
+	local_root->set_text(0, "Scene Groups");
+	local_root->set_icon(0, get_theme_icon(SNAME("PackedScene"), SNAME("EditorIcons")));
+	local_root->set_selectable(0, false);
+
+	for (Set<StringName>::Element *E = scene_groups[scene_root_node].front(); E; E = E->next()) {
+		if (!filter->get_text().is_subsequence_ofn(E->get())) {
+			continue;
 		}
-	}
 
-	if (name.is_empty()) {
-		renamed_group->set_text(0, selected_group);
-		error->set_text(TTR("Invalid group name."));
-		error->popup_centered();
-		return;
-	}
+		Node *n = node;
+		bool can_be_deleted = true;
 
-	renamed_group->set_text(0, name); // Spaces trimmed.
+		while (n) {
+			Ref<SceneState> ss = (n == EditorNode::get_singleton()->get_edited_scene()) ? n->get_scene_inherited_state() : n->get_scene_instance_state();
 
-	undo_redo->create_action(TTR("Rename Group"));
+			if (ss.is_valid()) {
+				int path = ss->find_node_by_path(n->get_path_to(node));
+				if (path != -1) {
+					if (ss->is_node_in_group(path, E->get())) {
+						can_be_deleted = false;
+					}
+				}
+			}
 
-	List<Node *> nodes;
-	scene_tree->get_nodes_in_group(selected_group, &nodes);
-	bool removed_all = true;
-	for (Node *node : nodes) {
-		if (_can_edit(node, selected_group)) {
-			undo_redo->add_do_method(node, "remove_from_group", selected_group);
-			undo_redo->add_undo_method(node, "remove_from_group", name);
-			undo_redo->add_do_method(node, "add_to_group", name, true);
-			undo_redo->add_undo_method(node, "add_to_group", selected_group, true);
+			n = n->get_owner();
+		}
+
+		TreeItem *item = tree->create_item(local_root);
+		item->set_cell_mode(0, TreeItem::CELL_MODE_CHECK);
+		item->set_editable(0, true);
+		item->set_checked(0, current_groups.find(E->get()) != nullptr);
+		item->set_text(0, E->get());
+		item->set_meta("__local", true);
+		item->set_meta("__name", E->get());
+		item->set_meta("__description", "");
+		if (can_be_deleted) {
+			//item->add_button(0, get_theme_icon(SNAME("Remove"), SNAME("EditorIcons")), DELETE_GROUP);
+			item->add_button(0, get_theme_icon(SNAME("ActionCopy"), SNAME("EditorIcons")), COPY_GROUP);
```

Comment:
Oh, I understood. It's for Instantiated Child Scenes. In original version, you cannot delete or copy groups from these nodes.
![image](https://user-images.githubusercontent.com/1756388/169314662-9cb8844f-f1cb-4b21-bd1a-9999dda5c496.png)

I added Lock icon for not editable groups. They aren't editable only for instantiated nodes.
 ![image](https://user-images.githubusercontent.com/1756388/169315169-12e02601-7702-4bd5-95a1-521cb93e962b.png)

I think what to do with Delete and Rename for these groups. They cannot be deleted and renamed, because they aren't part of current scene.